atintegrators/attrackfunc.c

```diff
@@ -15,11 +15,17 @@ void trackRFCavity(double *r_in, double le, double nv, double freq, double h, do
 {
     int c;
 
+    /* If nv is 0 and length is 0, then skip this whole loop (good for passive rf cavities
+        anyway if there is a cavity length, we have to loop through the particles
+    */
+
     if (le == 0) {
```

Comment:
Here you can use
if(nv==0.0) return;
to exit the function instead. This would also cover the case with le!=0 which is not the case in your implementation

Reply:
But if le is larger than 0 I don't want to exit the function, I want to apply the drift.